src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```diff
@@ -0,0 +1,224 @@
+package org.fogbowcloud.saps.engine.core.archiver.storage.swift;
+
+import static org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorageConstants.INPUTDOWNLOADING_FOLDER;
+import static org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorageConstants.PREPROCESSING_FOLDER;
+import static org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorageConstants.PROCESSING_FOLDER;
+import static org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorageConstants.SAPS_TASK_STAGE_DIR_PATTERN;
+
+import java.io.File;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorage;
+import org.fogbowcloud.saps.engine.core.archiver.storage.exceptions.PermanentStorageException;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+import org.fogbowcloud.saps.engine.utils.SapsPropertiesConstants;
+import org.fogbowcloud.saps.engine.utils.SapsPropertiesUtil;
+
+public class SwiftPermanentStorage implements PermanentStorage {
+
+    private final SwiftAPIClient swiftAPIClient;
+    private final Properties properties;
+    private final String nfsTempStoragePath;
+    private final String containerName;
+    private final boolean debugMode;
+
+    private static int MAX_ARCHIVE_TRIES = 1;
+    private static int MAX_SWIFT_UPLOAD_TRIES = 2;
+    private static final String SWIFT_STORAGE_TASK_DIR_PATTERN = "%s" + File.separator + "%s" + File.separator + "%s";
+
+    public static final Logger LOGGER = Logger.getLogger(SwiftPermanentStorage.class);
```

Comment:
the same comments made in the previous files apply in this one. e.g. why using properties deeper on the code? why not using the variables directly (after parse)?

Reply:
Added fixme.